client/descriptions.go

```diff
@@ -69,26 +72,52 @@ func (col CollectionDescription) IDString() string {
 	return fmt.Sprint(col.ID)
 }
 
-// GetFieldByID searches for a field with the given ID. If such a field is found it
+// GetFieldByName returns the field for the given field name. If such a field is found it
 // will return it and true, if it is not found it will return false.
-func (col CollectionDescription) GetFieldByID(id FieldID, schema *SchemaDescription) (FieldDescription, bool) {
-	for _, field := range schema.Fields {
-		if field.ID == id {
-			return field, true
+func (def CollectionDefinition) GetFieldByName(fieldName string) (FieldDescription, bool) {
```

Comment:
Okay, I did like having all the `GetFieldByName` next to each other, but that's partly because the Descriptions are all jumbled up in this one file anyway, and in a way its odd that Definition isn't.

Will let this discussion sit for a while and you/anyone-else talk some more before I try and make a choice RE its location :)

- [ ] Wait and see